eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -372,5 +373,49 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
     assert(new String(warn2.data.toArray).startsWith("invalid announcement sig"))
   }
 
+  test("establish transient connection") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val probe = TestProbe()
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(remoteNodeId, None)), Nil)
+    probe.send(peerConnection, message)
+    probe watch peerConnection
+    Thread.sleep(900)
+    assert(peerConnection.stateName === PeerConnection.CONNECTED)
+    Thread.sleep(200)
+    probe.expectTerminated(peerConnection, max = Duration.Zero)
+  }
+
+  test("keep using transient connection") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val probe = TestProbe()
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(remoteNodeId, None)), Nil)
+    probe.send(peerConnection, message)
+    probe watch peerConnection
+    Thread.sleep(900)
+    assert(peerConnection.stateName === PeerConnection.CONNECTED)
+    probe.send(peerConnection, message)
+    Thread.sleep(900)
+    assert(peerConnection.stateName === PeerConnection.CONNECTED)
+    probe.send(peerConnection, message)
+    Thread.sleep(900)
+    assert(peerConnection.stateName === PeerConnection.CONNECTED)
+    Thread.sleep(200)
+    probe.expectTerminated(peerConnection, max = Duration.Zero)
+  }
+
+  test("convert transient connection to persistent") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val probe = TestProbe()
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(remoteNodeId, None)), Nil)
+    probe.send(peerConnection, message)
+    Thread.sleep(900)
```

Comment:
Same, this test doesn't really need the `Thread.sleep`, we only want to send a channel message and then check that the connection is now persistent: I think you can access the actor state to directly verify the boolean, can't you?